lib/App/MintTag.pm

```diff
@@ -493,6 +495,42 @@ sub _maybe_delete_source_branches ($self, $step) {
   }
 }
 
+sub _maybe_cleanup_tags ($self, $step) {
+  return unless ($step->push_tag_to && $step->cleanup_tag_days);
+
+  my $remote = $step->push_tag_to;
+  my $prefix = $step->tag_prefix;
+
+  try {
+    my $tag_spec = "refs/tags/${prefix}*";
+    my $tag_list = run_git('for-each-ref', "--format=%(refname:short)\t%(creatordate:unix)", $tag_spec);
+
+    my @tags = map {; my ($tag, $ts) = split /\t/ ; { ts => $ts, 'tag' => $tag}  }
+    split /\r?\n/, $tag_list;
+
+    my $time = time;
+
+    my $max_ts = $time - ($step->cleanup_tag_days * 86400);
+
+    for my $chunk (@tags) {
+      if ($chunk->{ts} < $max_ts) {
+        $Logger->log(["Deleting old tag %s", $chunk->{tag}]);
+        run_git('push', $remote->name, ":refs/tags/$chunk->{tag}"  );
+      } else {
+        $Logger->log(["Keeping old tag %s", $chunk->{tag}]);
```

Comment:
This should, I think, be log_debug.  Otherwise you're going to emit N log lines for the N kept tags all the time, which seems like noise.  Also, if you use the all-in-one-batch from above, you could do the same to get a list and log them all at once.  But mostly: I wouldn't bother, even in a debug.

*Finally* (maybe):  you could use `List::MoreUtils::part` to split the list into keep/drop lists in one go, if you're feeling adventurous.  (I wouldn't bother, but it's nice to know it's out there.)